File: 204-count-primes/204-count-primes.java
class Solution {
    public int countPrimes(int n) {
 boolean[] arr = new boolean[n + 1];
for (int i = 1; i <= n; i++) {
    arr[i] = true;
}


for (int p = 2; p * p <= n; p++) {
    if (arr[p] == true) {
        for (int i = p * p; i <= n; i += p) {
            arr[i] = false;
        }
    }
}


int count = 0;
for (int i = 2; i < n; i++) {
    if (arr[i] == true) {
        count++;
    }
}
return count;
    }
}

File: 204-count-primes/204-count-primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        bool primes[n+1];
        memset(primes,true,sizeof(primes));
        
        // primes[0]=false;
        // primes[1]=false;
        for(int i=2;i*i<n;i++){
            if(primes[i]){
                for(int j=i*i;j<=n;j+=i){
                    primes[j]=false;
                }
            }
        }
        int cnt=0;
        for(int i=2;i<n;i++)if(primes[i])cnt++;
        return cnt;
    }
};